### ISSUE TITLE
Detach is not working

### ISSUE THREAD

**Issue by latot** [brave/kuchikiki#58]
Hi, I was trying to do very simple, detatch an element, but after do it, and check if was removed maybe seems not to be the case? this is because I'm also using ```eq``` to compare the result, just here is something wrong.

```rust
use kuchiki::traits::*;

fn main() {
    let html = kuchiki::parse_html().one(r#"<html><body><table></table></body></html>"#);
    let html2: kuchiki::NodeRef = html.clone();

    if let Some(element) = html.select_first("table").ok() {
        let as_node = element.as_node();
        as_node.detach();
    }

    let e = html.eq(&html2);
    println!("{e}");
}
```

It returns ```true```... when they are not equals, or maybe I did something bad?

Thx!

**Comment by latot** [comment]
Oks, there is other issues, but this is not the right one.
